fix: auto upgrade toast message

Raviguntakala opened the pull request:
### Problem
1. Upgrade toast notifications weren't showing because `upgrade()` was called in worker initialization before any clients connected to the event stream
2. When auto-update is disabled, users weren't notified that updates are available

### Solution
1. Added `checkUpgrade()` RPC method to worker that calls `upgrade()` in proper Instance context
2. TUI thread triggers `checkUpgrade()` via RPC after TUI starts, ensuring Instance is bootstrapped
3. Added `Installation.Event.UpdateAvailable` to notify users when auto-update is disabled without auto-updating them
4. Added toast notification handler in TUI to display manual update prompt when updates are available but auto-update is off

### Testing
- Verified auto-update complete toast notification appears when auto-update is enabled
- Verified manual update available toast notification appears when auto-update is disabled
- Both notifications display the correct version information

### Files Changed
- `packages/opencode/src/cli/cmd/tui/worker.ts` - Added checkUpgrade() RPC method with Instance.provide()
- `packages/opencode/src/cli/cmd/tui/thread.ts` - Call checkUpgrade() RPC after TUI starts
- `packages/opencode/src/installation/index.ts` - Added UpdateAvailable event definition
- `packages/opencode/src/cli/upgrade.ts` - Modified to publish UpdateAvailable event when auto-update is disabled
- `packages/opencode/src/cli/cmd/tui/app.tsx` - Added toast handler for UpdateAvailable event
- `packages/sdk/js/src/gen/types.gen.ts` - Regenerated SDK types to include UpdateAvailable event

![IMG_8329](https://github.com/user-attachments/assets/75c01a67-0296-41a8-b5bb-437c3a166dea)

<img width="1362" height="430" alt="Screenshot 2025-11-22 at 12 56 56 PM" src="https://github.com/user-attachments/assets/0e07a3cf-8ce7-410e-8f51-63348bd387a9" />

Raviguntakala committed:
fix: move upgrade call from worker to event stream handler

Move upgrade() from TUI worker initialization to /event endpoint handler.
The upgrade check now runs 1 second after a client connects to the event
stream, ensuring proper delivery of upgrade notifications.

rekram1-node commented:
To address that issue do you think we could update upgrade function to notify the user of an update if there is one but NOT auto update them (may need to create a new event for it)

rekram1-node commented on packages/opencode/src/server/server.ts:
```diff
@@ -1984,6 +1985,9 @@ export namespace Server {
               })
             })
+            setTimeout(() => {
+              upgrade()
```
hm this probably isn't the best spot for this tbh. The server is used in several types of environments (some people spawn these servers and connect multiple clients to it). Ideally we could have some setup w/ worker + tui thread so that once the tui is booted we run this

Raviguntakala replied:
Updated to use worker + TUI thread approach instead. Added `checkUpgrade()` RPC method to worker that's called from TUI thread after startup. This ensures it only runs once per TUI session and doesn't affect other server environments.

Raviguntakala committed:
feat: show toast notification for manual updates when auto-update disabled

Raviguntakala commented:
> To address that issue do you think we could update upgrade function to notify the user of an update if there is one but NOT auto update them (may need to create a new event for it)

Yes, I've added `Installation.Event.UpdateAvailable` that publishes when auto-update is disabled. It notifies users of the available version and prompts them to run `opencode upgrade` manually.
<img width="1362" height="430" alt="Screenshot 2025-11-22 at 12 56 56 PM" src="https://github.com/user-attachments/assets/d8723d14-1b85-4236-8bde-de8390c2ea41" />

Raviguntakala committed:
refactor: move upgrade check from server to worker RPC

Call upgrade() via worker.checkUpgrade() RPC from TUI thread instead of in server /event endpoint. This ensures upgrade runs in proper Instance context and avoids running on every client connection.

Raviguntakala committed:
refactor: remove unnecessary comments in TUI and upgrade functions

Raviguntakala committed:
Merge branch 'sst:dev' into dev

thdxr committed:
tui: keep assistant footer from crashing after compaction

Commit:
release: v1.0.101

Raviguntakala committed:
Merge branch 'sst:dev' into dev

Raviguntakala committed:
Merge branch 'dev' into dev

Raviguntakala committed:
fix: remove unnecessary blank lines in thread, worker, and upgrade files; add newline at end of package.json files
